client/my-sites/marketing/connections/services/google-drive.js

```diff
@@ -0,0 +1,93 @@
+import { last, isEqual } from 'lodash';
+import PropTypes from 'prop-types';
+import { SharingService, connectFor } from 'calypso/my-sites/marketing/connections/service';
+import { deleteStoredKeyringConnection } from 'calypso/state/sharing/keyring/actions';
+
+export class GoogleDrive extends SharingService {
+	static propTypes = {
+		// This foreign propTypes access should be safe because we expect all of them to be removed
+		// eslint-disable-next-line react/forbid-foreign-prop-types
+		...SharingService.propTypes,
+		deleteStoredKeyringConnection: PropTypes.func,
+	};
+
+	static defaultProps = {
+		...SharingService.defaultProps,
+		deleteStoredKeyringConnection: () => {},
+	};
+
+	createOrUpdateConnection = () => {};
+
+	/**
+	 * Deletes the passed connections.
+	 */
+	removeConnection = () => {
+		this.setState( { isDisconnecting: true } );
+		this.props.deleteStoredKeyringConnection( last( this.props.keyringConnections ) );
+	};
+
+	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
```

Comment:
I think it might be nice to not introduce legacy methods Calypso is actively trying to get rid of.

In this case, you'll probably want to use `componentDidUpdate`. There's just one caveat:  
Unlike `componentWillReceiveProps`, calling `this.setState` inside of `componentDidUpdate` will trigger a re-render of the component, so `componentDidUpdate` will be called yet again, so you need to watch out for loops but the way these conditions are organized, I think it should be mostly ok.

Reply:
I did change it to `componentDidUpdate` but then the linter complained about setting state, saying "Do not use setState in componentDidUpdate" for the three times it's used.

Reply:
The Facebook service does it in a different way, which I have copied. The rest of those services need to be updated to remove that unsafe function..